iox_query/src/exec/gapfill/algo/interpolate.rs

```diff
@@ -0,0 +1,545 @@
+//! Filling gaps with interpolated values.
+use std::sync::Arc;
+
+use arrow::{
+    array::{
+        Array, ArrayRef, Float64Array, Int64Array, PrimitiveArray, TimestampNanosecondArray,
+        UInt64Array,
+    },
+    datatypes::{ArrowPrimitiveType, DataType},
+};
+
+use crate::exec::gapfill::params::GapFillParams;
+
+use datafusion::{
+    error::{DataFusionError, Result},
+    scalar::ScalarValue,
+};
+
+use super::{AggrColState, Cursor, RowStatus, VecBuilder};
+
+/// [Cursor] methods that are related to interpolation.
+impl Cursor {
+    /// Create an Arrow array with gaps filled in between values
+    /// using linear interpolation.
+    pub(super) fn build_aggr_fill_interpolate(
+        &mut self,
+        params: &GapFillParams,
+        series_ends: &[usize],
+        input_time_array: &TimestampNanosecondArray,
+        input_aggr_array: &ArrayRef,
+    ) -> Result<ArrayRef> {
+        match input_aggr_array.data_type() {
+            DataType::Int64 => {
+                let input_aggr_array = input_aggr_array
+                    .as_any()
+                    .downcast_ref::<Int64Array>()
+                    .unwrap();
```

Comment:
As an FYI you can use https://docs.rs/arrow-array/35.0.0/arrow_array/cast/fn.as_primitive_array.html here too if you wanted slightly more concision

```suggestion
                let input_aggr_array = as_primitive_array<Int64Type>
                  (input_aggr_array);

```

Reply:
I like concision, I will do it, thank you.